### PR TITLE
Searching/ sorting of employees and pagination server side

### PR DESCRIPTION
**Pull request by ShivaniBansal1**

### CODE REVIEW

File: api/controllers/base.js [hunk 1]
```diff
@@ -11,17 +13,70 @@ class Base {
...
+            let queryObject = { $regex: req.params.searchBy, $options: 'i'};
+            const searchedRecords = await this.model.getAll({name: queryObject});
+            res.status(200).send({
+                success: true,
+                payload: {
+                    data : {
+                        searchedRecords
+                    },
+                    message: "list of searched records returned successfully!!"
+                }
```
Comment (sonalichawla16) [inline api/controllers/base.js:35]: maintain proper casing in message returning in payload and try to make it more meaningful

File: api/controllers/base.js [hunk 2]
```diff
@@ -11,17 +13,70 @@ class Base {
...
+            const pager = await pagination.paginate(recordList.length, page, pageSize);
+            const pageOfItems = recordList.slice(pager.startIndex, pager.endIndex + 1);
+            
+            res.status(200).send({
+                success : true,
+                data : {
+                    pager : pager,
+                    listOfData : pageOfItems,
+                    message : "list of data returned successfully!!"
+                }
```
Comment (sonalichawla16) [inline api/controllers/base.js:64]: maintain proper casing in message returning in payload and try to make it more meaningful

File: api/routes/index.js [hunk 3]
```diff
@@ -4,13 +4,17 @@ module.exports=(app) =>
     //Employee
     app.post('/api/employee', controller.employee.save);
     app.get('/api/employee/:id', controller.employee.get);
-    app.get('/api/employee', controller.employee.getAll);
+    app.get("/api/employees", (req, res)=>controller.employee.getAll(req, res));
+    app.get("/api/employeeBySearch/:searchBy", (req, res)=>controller.employee.searchRecord(req, res));
```
Comment (sonalichawla16) [inline api/routes/index.js:8]: make sure that we are using the same /api/***** format in all the APIs

File: api/controllers/base.js (hunk 1 above)
Comment (ShivaniBansal1) [reply api/controllers/base.js:35]: Okay.

File: api/controllers/base.js (hunk 2 above)
Comment (ShivaniBansal1) [reply api/controllers/base.js:64]: Okay

File: api/routes/index.js (hunk 3 above)
Comment (ShivaniBansal1) [reply api/routes/index.js:8]: Sure.

**Pull request merged**